test/test_jit.py

```diff
@@ -6217,6 +6217,49 @@ def test_index_put(target, indices, rhs):
 
         self.assertExpectedGraph(test_index_put.graph)
 
+    def test_tuple_indexing(self):
+        @torch.jit.script
+        def tuple_index(a):
+            if a:
+                b = (1, 2)
+            else:
+                b = (0, 2)
+            return b[-2], b[1]
+
+        self.assertExpectedGraph(tuple_index.graph)
+        self.assertEqual(tuple_index(torch.tensor([1])), (1, 2))
+        self.run_pass('lower_all_tuples', tuple_index.graph)
+        self.assertEqual(tuple_index(torch.tensor([1])), (1, 2))
```

Comment:
We are still not running those ops even *once* in our tests...

Reply:
I'm running it on line 6230

Reply:
This is the assertEqual check on line 6230

Reply:
Mutating the graph after you have already called `tuple_index` will not cause it to use the mutated graph. Graphs are copied by the graph executor and specialized.